adding putmask and its test

ShreyanshBardia opened the pull request:
closes #18359

bipinKrishnan commented:
Thanks for the contribution @ShreyanshBardia :slightly_smiling_face: . Can you fix the failing test for paddle backend?
Also, it would be nice if you install pre-commit in your system, so that the lint checks are done correctly before pushing. Thanks

bipinKrishnan commented:
ivy-gardener

ShreyanshBardia commented:
@bipinKrishnan surely will run precommit beforehand. As for the paddle test case, I am not sure if that is being caused by my PR. A similar test case was failing for #18334, but it appears to be unrelated to the PR. Could confirm if this is the same case, if not I'll make the required changes.

ShreyanshBardia commented:
Hi @bipinKrishnan can you check this once

bipinKrishnan commented on ivy/functional/frontends/numpy/indexing_routines/inserting_data_into_arrays.py:
```diff
@@ -28,3 +29,12 @@ def fill_diagonal(a, val, wrap=False):
+        return
+    if values.size != a.size:
+        values = np_frontend.resize(values, a.shape)
+    a[mask] = values[mask]
```
I think you forgot the `return` for this function

ShreyanshBardia replied:
According to the documentation, the function doesn't return anything rather changes the array in place. [Here](https://numpy.org/doc/stable/reference/generated/numpy.putmask.html) is the relevant link

ivy-seed commented:
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

ShreyanshBardia commented:
Hi @bipinKrishnan, could you please take a look at this.

bipinKrishnan commented:
Hey @ShreyanshBardia , can you fix the failing the test for the function you added, currently failing for the paddle backend

ShreyanshBardia commented:
Hi @bipinKrishnan, as I mentioned earlier I think the failure is unrelated to the PR, a similar test was failing here https://github.com/unifyai/ivy/pull/18334#pullrequestreview-1524594910 which turned out to be unrelated. Would be glad if you could confirm that

bipinKrishnan commented:
Ok, let me check this

ivy-seed commented:
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

ShreyanshBardia commented:
Hi @bipinKrishnan, can you take a look at the PR now, I have updated the branch and also updates the test function, since putmask doesn't return anything and changed inplace it is better to test it directly using `helpers.value_test`

ShreyanshBardia commented:
@bipinKrishnan It seems an old docker image is being used to test and since I updated my branch, a new dependency that was recently added `dill` is causing the tests to fail

bipinKrishnan commented:
Ok, I'll let you know once this is fixed and let's see how the testing goes.

bipinKrishnan commented:
Also it would be great if you could configure pre-commit in your system to avoid lint test fails. Thanks!

bipinKrishnan commented:
ivy-gardener

bipinKrishnan commented:
> @bipinKrishnan It seems an old docker image is being used to test and since I updated my branch, a new dependency that was recently added `dill` is causing the tests to fail

The team is looking into this issue, let's see how the testing goes once this is fixed. Thanks!

bipinKrishnan commented:
Hey @ShreyanshBardia , the issue with `dill` is fixed now, can you rebase on top of main and try again?

ShreyanshBardia commented:
Hi, I do have pre-commit configured but I guess this might have been caused when I was resolving merge conflicts using web browser. Apologies for that. It looks like tests ran and were successful can you check @bipinKrishnan

bipinKrishnan commented on ivy/functional/frontends/numpy/indexing_routines/inserting_data_into_arrays.py:
```diff
@@ -31,6 +31,15 @@ def fill_diagonal(a, val, wrap=False):
 
 
+@to_ivy_arrays_and_back
+def putmask(a, mask, values, /):
```
You could remove the `/` as it doesn't make sense because there are no kwargs after that

ShreyanshBardia replied:
Oh correct, thanks for catching this, I have updated it correctly now @bipinKrishnan

bipinKrishnan commented:
ivy-gardener

ShreyanshBardia commented:
@bipinKrishnan can you take a look
